### ISSUE TITLE
When asserting equality of two values, instead of using `EXPECT_TRUE(x == y)` in gtest, use `EXPECT_EQ(x, y)`, to improve logging and debuggability of test failures.

### ISSUE THREAD

**Issue by ahsonkhan** [Azure/azure-sdk-for-cpp#2057]
https://github.com/Azure/azure-sdk-for-cpp/search?q=EXPECT_TRUE

For example:
https://github.com/Azure/azure-sdk-for-cpp/blob/63b2d21bc65a4348cad5fd69cecbe353f74b7120/sdk/core/azure-core/test/ut/nullable.cpp#L15
https://github.com/Azure/azure-sdk-for-cpp/blob/c5ddbbc430790fb8718dab7df792633c63f33d19/sdk/core/azure-core/test/ut/context.cpp#L43
https://github.com/Azure/azure-sdk-for-cpp/blob/b7de24483a8cb838484e6a1d2d2dbe895e582f04/sdk/core/azure-core/test/ut/uuid.cpp#L14

From https://github.com/Azure/azure-sdk-for-cpp/pull/2037#discussion_r608129450

cc @antkmsft

**Comment by ahsonkhan** [comment]
If a test fails today, we get the following:
```text
 Uuid.Basic
   Source: uuid.cpp line 12
   Duration: < 1 ms

  Message: 
    
    ..\..\..\sdk\core\azure-core\test\ut\uuid.cpp(14): error: Value of: uuid.ToString().length() == 35
      Actual: false
    Expected: true
```

Using `EXPECT_EQ` would get us a more meaningful output:
```text
 Uuid.Basic
   Source: uuid.cpp line 12
   Duration: 1 ms

  Message: 
    
    ..\..\..\sdk\core\azure-core\test\ut\uuid.cpp(14): error: Expected equality of these values:
      uuid.ToString().length()
        Which is: 36
      35
```

**Comment by antkmsft** [comment]
Recommendation: review all `EXPECT_TRUE` and `EXPECT_FALSE` - replace with `EXPECT_EQ`, `EXPECT_NE`, `EXPECT_GT`, `EXPECT_GE` etc.

**Comment by ahsonkhan** [comment]
Not necessarily all though. Some of them are expecting/asserting the bool returning condition of an operation. Those can be left as is. It is only when you are explicitly comparing two non-bool values, where you would use the alternatives you suggested.

**Comment by antkmsft** [comment]
I agree, there could be rare exceptions; But the majority of cases does need this conversion.

**Comment by Jinming-Hu** [comment]
storage sdk done in https://github.com/Azure/azure-sdk-for-cpp/pull/2347. Removing storage label

**Comment by antkmsft** [comment]
Resolved in #4457.
